recipes/openvdb/8.1.0/conanfile.py

```diff
@@ -0,0 +1,202 @@
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "with_blosc": [True, False],
+        "with_zlib": [True, False],
+        "with_log4cplus": [True, False],
+        "simd": [None, "SSE42", "AVX"],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "with_blosc": True,
+        "with_zlib": True,
+        "with_log4cplus": False,
+        "simd": None,
+    }
+
+    _cmake = None
+
+    _compilers_min_version = {
+        "msvc": "19.10",
+        "Visual Studio": "15",  # Should we check toolset?
+        "gcc": "6.3.1",
+        "clang": "3.8",
+        "apple-clang": "3.8",
+        "intel": "17",
+    }
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def _check_compilier_version(self):
+        compiler = str(self.settings.compiler)
+        version = tools.Version(self.settings.compiler.version)
+        if version < self._compilers_min_version[compiler]:
+            raise ConanInvalidConfiguration("%s requires a %s version greater than %s" % (self.name, compiler, self._compilers_min_version[compiler]))
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+            if self.settings.compiler == "Visual Studio" and "MT" in self.settings.compiler.runtime:
+                raise ConanInvalidConfiguration("Visual Studio build for shared library with MT runtime is not supported")
+
+        if self.settings.compiler.cppstd:
+            tools.check_min_cppstd(self, 14)
+        if self.settings.arch not in ("x86", "x86_64"):
+            if self.options.simd:
+                raise ConanInvalidConfiguration("Only intel architectures support SSE4 or AVX.")
+        self._check_compilier_version()
+
+    def requirements(self):
+        self.requires("boost/1.76.0")
+        self.requires("tbb/2020.3")
+        if self.options.with_zlib:
+            self.requires("zlib/1.2.11")
+        if self.options.with_blosc:
+            self.requires("c-blosc/1.20.1")
+        if self.options.with_log4cplus:
+            self.requires("log4cplus/2.0.5")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version], strip_root=True, destination=self._source_subfolder)
+
+    def _patch_sources(self):
+        # Remove FindXXX files from OpenVDB. Let Conan do the job
+        tools.remove_files_by_mask(os.path.join(self._source_subfolder, "cmake"), "Find*")
+        with open("FindBlosc.cmake", "w") as f:
+            f.write(
+                """find_package(c-blosc)
+if(c-blosc_FOUND)
+    add_library(blosc INTERFACE)
+    target_link_libraries(blosc INTERFACE c-blosc::c-blosc)
+    add_library(Blosc::blosc ALIAS blosc)
+endif()
+"""
+            )
```

Comment:
Use [cmake_find_package](https://docs.conan.io/en/latest/integrations/build_system/cmake/cmake_find_package_generator.html) Generator instead.
```suggestion
```

Reply:
If this wants custom targets.... This could be moved to the cmake_wrapper too?

Reply:
The problem is that the conan recipe for c-blosc names the target as `c-blosc` while the c-blosc library generates it as `blosc`. I had in mind to update the c-blosc recipe but I wasn't sure because it might break many things.

The find_package wrapper to allow different names comes from the previous openvdb version recipe [here](https://github.com/conan-io/conan-center-index/pull/5257#discussion_r615432593)

Reply:
The project does not (obviously) install any CMake targets... which is why the Conan ones may not match 
https://github.com/Blosc/c-blosc/blob/f4ef2f357900c515a100320ef0494da74bc687a7/CMakeLists.txt#L339